CentralityiOS/CentralityiOS/ViewControllers/UsageDashboardViewController.m

```diff
@@ -0,0 +1,177 @@
+    
+    
+    
+    [self.pieChart animateWithXAxisDuration:1.4 easingOption:ChartEasingOptionEaseOutBack];
+}
+
+- (void)loadChartWithCategories{
+    [[CentralityHelpers queryForUsersCategories] findObjectsInBackgroundWithBlock:^(NSArray * _Nullable categoriesFromQuery, NSError * _Nullable error) {
+        self.userCategories = [categoriesFromQuery mutableCopy];
+        [self updateChartData];
+    }];
+}
+
+- (void)setupPieChartView:(PieChartView *)chartView
+{
+    chartView.usePercentValuesEnabled = YES;
+    chartView.drawSlicesUnderHoleEnabled = NO;
+    chartView.holeRadiusPercent = 0.58;
+    chartView.transparentCircleRadiusPercent = 0.61;
+    chartView.chartDescription.enabled = NO;
+    [chartView setExtraOffsetsWithLeft:5.f top:10.f right:5.f bottom:5.f];
+    
+    chartView.drawCenterTextEnabled = YES;
+    
+    NSMutableParagraphStyle *paragraphStyle = [[NSParagraphStyle defaultParagraphStyle] mutableCopy];
+    paragraphStyle.lineBreakMode = NSLineBreakByTruncatingTail;
+    paragraphStyle.alignment = NSTextAlignmentCenter;
+    
+    NSMutableAttributedString *centerText = [[NSMutableAttributedString alloc] initWithString:@"Tasks\nby Category"];
+    [centerText setAttributes:@{
+        NSFontAttributeName: [UIFont fontWithName:@"HelveticaNeue-Light" size:11.f],
+        NSParagraphStyleAttributeName: paragraphStyle
+    } range:NSMakeRange(0, centerText.length)];
+    chartView.centerAttributedText = centerText;
+    
+    chartView.drawHoleEnabled = YES;
+    chartView.rotationAngle = 0.0;
+    chartView.rotationEnabled = YES;
+    chartView.highlightPerTapEnabled = YES;
+    
+    ChartLegend *l = chartView.legend;
+    l.horizontalAlignment = ChartLegendHorizontalAlignmentRight;
+    l.verticalAlignment = ChartLegendVerticalAlignmentTop;
+    l.orientation = ChartLegendOrientationVertical;
+    l.drawInside = NO;
+    l.xEntrySpace = 7.0;
+    l.yEntrySpace = 0.0;
+    l.yOffset = 0.0;
+}
+
+- (void)updateChartData
+{
+    if (self.shouldHideData)
+    {
+        self.pieChart.data = nil;
+        return;
+    }
+    
+    [self setDataCount:self.userCategories.count];
+}
+
+- (void)setDataCount:(NSUInteger)count
+{
+    NSMutableArray *values = [[NSMutableArray alloc] init];
+    for (int i = 0; i < count; i++)
+    {
+        CategoryObject* currentCategory = self.userCategories[i % self.userCategories.count];
+        [values addObject:[[PieChartDataEntry alloc] initWithValue:currentCategory.numberOfTasksInCategory label:currentCategory.categoryName icon: [UIImage imageNamed:@"icon"]]];
+    }
+    
+    PieChartDataSet *dataSet = [[PieChartDataSet alloc] initWithEntries:values label:@"Tasks by Category"];
+    
+    dataSet.drawIconsEnabled = NO;
+    
+    dataSet.sliceSpace = 2.0;
+    dataSet.iconsOffset = CGPointMake(0, 40);
+    
+    // Category Colors can be customized here
+    
+    NSMutableArray *colors = [[NSMutableArray alloc] init];
+    [colors addObjectsFromArray:ChartColorTemplates.vordiplom];
```

Comment:
`vordiplom: German undergraduate degree`

that's a word i didn't know before today.

Reply:
I'm just learning this now! Really interesting